android/FirebaseContinue/library/src/main/java/com/firebasecontinue/FirebaseContinue.java

```diff
@@ -0,0 +1,174 @@
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ServerValue;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+/**
+ * The Firebase Continue for Android library.
+ *
+ * Firebase Continue enables mobile developers to easily integrate activity transitioning
+ * from their mobile apps to the web by way of Chrome extensions.
+ * For more details, see: https://github.com/firebase/firebase-continue.
+ *
+ * Please see the usage instructions in the relevant README file(s).
+ * There is also more specific documentation within the library itself below.
+ *
+ * TODO: Add unit tests, including tests while the app/Firebase is offline.
+ */
+public final class FirebaseContinue {
+
+    /**
+     * Attempts to asynchronously broadcast an Activity (codified as a URL) within an application
+     * that the currently signed in user may wish to continue elsewhere (in the immediate future)
+     * to all potential clients (i.e. Chrome extension(s)) which could allow the user to do so by
+     * opening said URL.
+     *
+     * Note that, by design, only the most recently successfully broadcast Activity (for a given
+     * application) could possibly be relevant to the user. The Firebase Continue database rules
+     * and libraries enforce this. For more details, please see the relevant README file(s).
+     *
+     * @param activityUrl The URL which, if the current user were to navigate to,
+     *                    would allow the user to continue their Activity.
+     * @param applicationName The name of the application, as defined in the Firebase Realtime
+     *                        Database rules for Firebase Continue, that the user's Activity is
+     *                        within.
+     * @return A Task which, upon completion, signals whether or not the Firebase Continue Activity
+     * with the provided URL (within the context of the application with the provided appName)
+     * was successfully broadcast to all potential clients which could allow the user to continue
+     * the Activity, or that the broadcast failed. To understand how to use Tasks,
+     * see: https://developers.google.com/android/guides/tasks.
+     */
+    public static Task<Void> broadcastActivityToContinue(final String activityUrl,
+                                                         final String applicationName) {
+        // Use a chain of Tasks to eventually set the value for the current user of the most
+        // recent Activity they may wish to continue within the application.
+        return Tasks.call(new Callable<DatabaseReference>() {
+            @Override
+            public DatabaseReference call() throws Exception {
+                // First, ensure the inputs could be valid.
+                if (activityUrl == null) {
```

Comment:
I'd say we should be even stricter and disallow null or all-whitespace for both activityUrl and applicationName